model/artifact/db.go

```diff
@@ -104,6 +114,45 @@ func (e Entry) Upsert() error {
 	return err
 }
 
+func (e Entry) Update() error {
+	err := db.Update(
+		Collection,
+		bson.M{
+			TaskIdKey:    e.TaskId,
+			TaskNameKey:  e.TaskDisplayName,
+			BuildIdKey:   e.BuildId,
+			ExecutionKey: e.Execution,
+		},
+		bson.M{
+			"$set": bson.M{
+				FilesKey: e.Files,
+			},
+			"$setOnInsert": bson.M{
+				ExecutionKey: e.Execution,
+			},
+		},
+	)
+	if err != nil && e.Execution == 0 {
```

Comment:
tries again if it fails and the execution was 0 but wasnt sure which error to return

Reply:
Rather than running two Updates, can we case ahead and check if execution == 0, then match on `{"$or": {"ExecutionKey: {$exists: false}, ExecutionKey: 0}` (something like https://github.com/evergreen-ci/evergreen/blob/39faa9305d7f6447447544fc423240a670dbd5fd/model/distro/db.go#L149)?

Reply:
done